data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/mapping/AbstractLogstashPluginAttributesMapper.java

```diff
@@ -52,14 +54,17 @@
                                     dataPrepperAttributeName.substring(1), !(Boolean) logstashAttribute.getAttributeValue().getValue()
                             );
                         }
+                        else if(defaultSettings.containsKey(logstashAttributeName)) {
```

Comment:
The default settings should be used when there is no attribute value. This condition is in a loop which iterates over the existing attribute values. So it isn't the right place to have this.

At the end of this function, there should be another loop over `defaultSettings`. It should then see if the setting was previously set. If it was not already set, then it should be set with the default.